src/molecule/config.py

```diff
@@ -230,7 +230,7 @@ def _get_driver_name(self):
         driver_from_state_file = self.state.driver
         driver_from_cli = self.command_args.get("driver_name")
 
-        if driver_from_state_file:
+        if driver_from_state_file and driver_from_state_file in api.drivers():
```

Comment:
We should move that check to the end of the function and fail the Molecule run if the selected driver is missing.

Reply:
Done.

I added the check at the end:
`if driver_from_state_file and not driver_name in api.drivers()`

One could just check `if driver_name in api.drivers()` but it seems to me that the other cases (_input via shell argument and config_) already get validated.
Also - this way the user get's to know that the state-file is the information-source and where it is located.

Reply:
I wonder if we should just prompt the user to clean the cache in that case with an interactive prompt. WDYT? (obviously that this would appear only on tty sessions).

Reply:
Yeah - I believe CI wouldn't run into those problems as the cache/runner instance has a short lifetime.

May I ask completely unbiased => what is the state file even used for?
As far as I've seen its only reference inside molecule's source is this one.
`config.py -> self.state -> self.state.driver`
Have I overlooked something?